src/libs/actions/TaxRate.ts

```diff
@@ -0,0 +1,130 @@
+import Onyx from 'react-native-onyx';
+import * as API from '@libs/API';
+import type {CreatePolicyTaxParams} from '@libs/API/parameters';
+import {WRITE_COMMANDS} from '@libs/API/types';
+import CONST from '@src/CONST';
+import * as ErrorUtils from '@src/libs/ErrorUtils';
+import ONYXKEYS from '@src/ONYXKEYS';
+import type {TaxRate, TaxRates} from '@src/types/onyx';
+import type {PendingAction} from '@src/types/onyx/OnyxCommon';
+import type {OnyxData} from '@src/types/onyx/Request';
+
+/**
+ * Get tax value with percentage
+ */
+function getTaxValueWithPercentage(value: string): string {
+    return `${value}%`;
+}
+
+function covertTaxNameToID(name: string) {
+    return `id_${name.toUpperCase().replaceAll(' ', '_')}`;
+}
+
+/**
+ * Get new tax ID
+ */
+function getNextTaxID(name: string, taxRates?: TaxRates): string {
+    const newID = covertTaxNameToID(name);
+    if (!taxRates?.[newID]) {
+        return newID;
+    }
+
+    // If the tax ID already exists, we need to find a unique ID
+    let nextID = 1;
+    while (nextID < 100) {
```

Comment:
NAB but why 100? Shouldn't we continue this until `taxRates?.[newID]` is false?

Reply:
I thought it would be nice to add some high limit. But right, I've changed it